manager/state/raft/storage/snapwrap.go

```diff
@@ -93,3 +102,57 @@ func (o originalSnap) New(dirpath string) Snapshotter {
 
 // OriginalSnap is the original `snap` package as an implemntation of the SnapFactory interface
 var OriginalSnap SnapFactory = originalSnap{}
+
+// MigrateSnapshot reads the latest existing snapshot from one directory, encoded one way, and writes
+// it to a new directory, encoded a different way
+func MigrateSnapshot(oldDir, newDir string, oldFactory, newFactory SnapFactory) error {
+	// use temporary snaphot directory so initialization appears atomic
+	oldSnapshotter := oldFactory.New(oldDir)
+	snapshot, err := oldSnapshotter.Load()
+	switch err {
+	case snap.ErrNoSnapshot: // if there's no snapshot, the migration succeeded
```

Comment:
Shouldn't we still create the directory in this case (and leave it empty)? That will avoid future migrations, and I think the snapshotter expects the directory to exist.